pkg/metrics/metrics.go

```diff
@@ -0,0 +1,36 @@
+package metrics
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+type Server struct {
+	httpServer *http.Server
+	logger     log.Logger
+}
+
+func NewMetricsServer(logger log.Logger, port int) *Server {
+	mux := http.NewServeMux()
+	mux.Handle("/metrics", promhttp.Handler())
+
+	return &Server{
+		logger: logger,
+		httpServer: &http.Server{
+			Addr:    fmt.Sprintf(":%d", port),
```

Comment:
see https://github.com/grafana/pdc-agent/pull/100/files#r1638826052 - this should just take an address directly:

```suggestion
func NewMetricsServer(logger log.Logger, addr string) *Server {
	mux := http.NewServeMux()
	mux.Handle("/metrics", promhttp.Handler())

	return &Server{
		logger: logger,
		httpServer: &http.Server{
			Addr:    addr,
```